src/components/UploadZzal/ImageUpload.tsx

```diff
@@ -0,0 +1,115 @@
+import { Fragment, useRef, useState, DragEvent, ChangeEvent } from "react";
+import { Upload } from "lucide-react";
+import { cn } from "@/utils/tailwind";
+import ZzalCard from "@/components/common/ZzalCard";
+
+type DivDragEvent = DragEvent<HTMLDivElement>;
+interface Props {
+  onChange: (file: File | null) => void;
+}
+
+const ImageUpload = ({ onChange }: Props) => {
+  const [previewUrl, setPreviewUrl] = useState<string | null>(null);
+  const [dragging, setDragging] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const handleClickDeleteButton = () => {
+    onChange(null);
+    setPreviewUrl(null);
+  };
+
+  const handleFileChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const { files } = event.target;
+    if (files) {
+      const [changedFile] = files;
+      onChange(changedFile);
+      if (changedFile) {
+        setPreviewUrl(URL.createObjectURL(changedFile));
+      }
+    }
+  };
+
+  const handleChooseFile = () => {
+    if (inputRef.current) {
+      inputRef.current.click();
+    }
+  };
+
+  const handleDragEnter = (event: DivDragEvent) => {
+    event.preventDefault();
+
+    const { items } = event.dataTransfer;
+    if (items && items.length > 0) {
```

Comment:
```suggestion
    if (items?.length > 0) {
```
옵셔널 체이닝 연산자를 이용해서 좀 더 간결하게 작성해줄 수 있을 것 같아요:)